clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -75,6 +74,7 @@
     private final int autoCommitIntervalMs;
     private final ConsumerInterceptors<?, ?> interceptors;
     private final boolean excludeInternalTopics;
+    private final AtomicInteger pendingAsyncCommits;
```

Comment:
In order to use `ConsumerNetworkClient.pendingRequestCount(Node)`, we need to know the coordinator. But we don't want to wait to find the coordinator if there are no pending commits. If `pendingAsyncCommits` > 0, close waits  upto `timeout` to find coordinator and wait for `ConsumerNetworkClient.pendingRequestCount(coordinator)`  to become zero. We could use `ConsumerNetworkClient.pendingRequestCount()` instead to wait for  pending requests to all nodes, but I thought that would include fetch requests as well and I wanted to avoid that.